enterprise/service/license.go

```diff
@@ -44,7 +44,7 @@ func NewLicenseService(mode common.ReleaseMode, store *store.Store) (*LicenseSer
 func (s *LicenseService) StoreLicense(patch *enterpriseAPI.SubscriptionPatch) error {
 	if patch.License != "" {
 		if _, err := s.parseLicense(patch.License); err != nil {
-			return nil
+			return err
```

Comment:
Nolint

Reply:
Here, this place is correct, but we should convert common.Invalid error to bad request error in server/subscription.go, otherwise we are always returning internal errors.